src/EnergyPlus/ZoneAirLoopEquipmentManager.cc

```diff
@@ -192,13 +196,6 @@ namespace ZoneAirLoopEquipmentManager {
         Array1D_bool lNumericBlanks(2);   // Logical array, numeric field input BLANK = .TRUE.
         bool DualDuctRecircIsUsed;        // local temporary for deciding if recirc side used by dual duct terminal
```

Comment:
So, this group of seemingly small arrays suck up that much time being needlessly allocated? That's incredible. And these local arrays aren't even necessary. There's `state.dataIPShortCut->cAlphaArgs` etc. available to use. Not suggesting you need to change that here, but something to watch for elsewhere.

Reply:
Be careful! There is a fourPipeBeamFactory call at line 417 that will overwrite the dataIPShortCut variables. So I would think these locals are needed. The only time dataIPShortCut variables can be used is if the getInput in question has no calls to other getInput functions that use the dataIPShortCut variables.

Reply:
Sorry @rraustad, I'm confused. Where are you looking?

Here? https://github.com/NREL/EnergyPlus/blob/68b20c86427b300a29dedb8fd594716320b06a0d/src/EnergyPlus/ZoneAirLoopEquipmentManager.cc#L414-L418

Reply:
> So, this group of seemingly small arrays suck up that much time being needlessly allocated? That's incredible. 

They do! I think I mentioned in the most recent 10X "lesson" that `std::vector` and `ArrayXD` should not be used as local variables unless absolutely necessary because of the overhead of calling constructors (malloc) and destructors (free).

Reply:
I was looking at develop. Yes, that call will getInput the 4pipebeam and overwrite those dataIPShortCut variables mentioned. Low level models that do not do any mining of data can use dataIPShortCut variables (e.g., state.dataIPShortCut->cAlphaArgs). Higher level models that get data from other models usually need to use locals for the getObjectItem call (e.g., Array1D_string AlphArray(5)). This was over 10 years ago so it's not on anyone's radar. Once we switch over to json all these arrays go away.

Reply:
As @mjwitte said, there are no changes needed regarding this discussion.

Reply:
> Be careful! There is a fourPipeBeamFactory call at line 417 that will overwrite the dataIPShortCut variables. So I would think these locals are needed. The only time dataIPShortCut variables can be used is if the getInput in question has no calls to other getInput functions that use the dataIPShortCut variables.

Put a pin in this. Re-entrant code is not a good design pattern. Re-entrant code with variables that are effectively global is really not a good design pattern. This should be "flattened out" somehow.